### ISSUE TITLE
Error compiling threads.lisp

### ISSUE THREAD

**Issue by Eldred777** [nobody-famous/alive#85]
Hello, I am having trouble getting the LSP to work, it throws an error about not being able to compile the threads.lisp file. I have tried navigating directly to the folder containing threads.lisp and compiling there from SBCL, but it throws the same error. Not really sure where to go from here.

Running asdf version 3.3.5.8 and am on Windows 10. Have also tried uninstalling the whole extension and reinstalling, and reinstalling my whole SBCL installation. All required packages are installed, I think. 

The entire output from Alive LSP follows:
```
This is SBCL 2.2.4, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
; compiling file "c:/Users/ciara/.vscode/extensions/rheller.alive-0.3.2/out/alive-lsp/v0.1.0/nobody-famous-alive-lsp-0c3b9ce/src/compat/sbcl/threads.lisp" (written 03 JUN 2022 12:10:21 AM):
; 
; caught ERROR:
;   READ error during COMPILE-FILE:
;   

; compilation aborted after 0:00:00.001
;     Lock on package SB-THREAD violated when interning THREAD-OS-THREAD while in
;     package ALIVE/SBCL/THREADS.
;   See also:
;     The SBCL Manual, Node "Package Locks"
;   
;     (in form starting at line: 9, column: 0, position: 143)

debugger invoked on a UIOP/LISP-BUILD:COMPILE-FILE-ERROR in thread
#<THREAD "main thread" RUNNING {1005958003}>:

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [RETRY                        ] Retry
                                     compiling #<CL-SOURCE-FILE "alive-lsp/src/compat/sbcl/threads" "file-type">.
  1: [ACCEPT                       ] Continue, treating
                                     compiling #<CL-SOURCE-FILE "alive-lsp/src/compat/sbcl/threads" "file-type">
                                     as having been successful.
  COMPILE-FILE-ERROR while
  compiling #<CL-SOURCE-FILE "alive-lsp/src/compat/sbcl/threads" "file-type">
  2:                                 Retry ASDF operation.
  3: [CLEAR-CONFIGURATION-AND-RETRY] Retry ASDF operation after resetting the
                                     configuration.
  4: [CONTINUE                     ] Ignore runtime option --eval "(asdf:load-system :alive-lsp)".
  5: [ABORT                        ] Skip rest of --eval and --load options.
  6:                                 Skip to toplevel READ/EVAL/PRINT loop.
  7: [EXIT                         ] Exit SBCL (calling #'EXIT, killing the process).

(UIOP/LISP-BUILD:CHECK-LISP-COMPILE-RESULTS NIL T T "~/asdf-action::format-action/" ((#<ASDF/LISP-ACTION:COMPILE-OP > . #<ASDF/LISP-ACTION:CL-SOURCE-FILE "alive-lsp/src/compat/sbcl/threads" "file-type">)))
   source: (ERROR 'COMPILE-FILE-ERROR :CONTEXT-FORMAT CONTEXT-FORMAT
                  :CONTEXT-ARGUMENTS CONTEXT-ARGUMENTS)
0] 
```

**Comment by nobody-famous** [comment]
I dug around a little bit and apparently I got lucky in avoiding this. I primarily develop on Windows, but I've been using SBCL 2.0.0. Apparently, that's the last version that works. The rest of the 2.0 releases don't even have a Windows binary and the 2.1 release I tried also gave the error.

I have a fix for it, which'll go into the next release I put up. In the meantime, the simplest solution would be to downgrade to SBCL 2.0.0. The other option would be to clone the LSP source, point your workspace to it by adding this to your settings.json

"alive.lsp.install.path": "<wherever you put it>"

Then change src/compat/sbcl/threads.lisp and remove the #+/-win32 stuff so that get-thread-id returns the thread name.

I'll update this issue when I get the new release up.
